manager/controllers/app/blueprint_controller.go

```diff
@@ -109,6 +108,9 @@ func (r *BlueprintReconciler) removeFinalizers(ctx context.Context, cfg *action.
 			return client.IgnoreNotFound(err)
 		}
 	}
+	if environment.IsNPEnabled() {
```

Comment:
It should be done only if the finalizers are present, similar to deleteExternalResources.